tests/testthat/test-selections.R

```diff
@@ -0,0 +1,213 @@
+              new_role = "predictor") %>%
+  update_role(HHV, new_role = "outcome") %>%
+  update_role(sample, new_role = "id variable") %>%
+  update_role(dataset, new_role = "splitting indicator")
+info4 <- summary(rec4)
+
+test_that('simple role selections', {
+  expect_equal(
+    recipes_eval_select(quos = quos(all_predictors()), data = okc, info = info1),
+    setNames(nm = info1$variable)
+  )
+  expect_equal(
+    recipes_eval_select(quos = quos(all_outcomes()), data = okc, info = info1),
+    setNames(nm = character())
+  )
+  expect_equal(
+    recipes_eval_select(quos = quos(all_outcomes()), data = biomass, info = info4),
+    setNames(nm = "HHV")
+  )
+  expect_equal(
+    recipes_eval_select(
+      quos = quos(has_role("splitting indicator")),
+      data = biomass,
+      info = info4
+    ),
+    setNames(nm = "dataset")
+  )
+})
+
+test_that('simple type selections', {
+  expect_equal(
+    recipes_eval_select(quos = quos(all_numeric()), data = okc, info = info1),
+    setNames(nm = c("age", "height"))
+  )
+  expect_equal(
+    recipes_eval_select(quos = quos(has_type("date")), data = okc, info = info1),
+    setNames(nm = "date")
+  )
+  expect_equal(
+    recipes_eval_select(quos = quos(all_nominal()), data = okc, info = info1),
+    setNames(nm = c("diet", "location", "Class"))
+  )
+})
+
+test_that('simple name selections', {
+  expect_equal(
+    recipes_eval_select(quos = quos(matches("e$")), data = okc, info = info1),
+    setNames(nm = c("age", "date"))
+  )
+  expect_equal(
+    recipes_eval_select(quos = quos(contains("gen")), data = biomass, info = info4),
+    setNames(nm = c("hydrogen", "oxygen", "nitrogen"))
+  )
+  expect_equal(
+    recipes_eval_select(quos = quos(contains("gen"), -nitrogen), data = biomass, info = info4),
+    setNames(nm = c("hydrogen", "oxygen"))
+  )
+  expect_equal(
+    recipes_eval_select(quos = quos(date, age), data = okc, info = info1),
+    setNames(nm = c("date", "age"))
+  )
+  expect_equal(
+    recipes_eval_select(quos = quos(-age, date), data = okc, info = info1),
+    setNames(nm = c("diet", "height", "location", "date", "Class"))
+  )
+  expect_equal(
+    recipes_eval_select(quos = quos(date, -age), data = okc, info = info1),
+    setNames(nm = "date")
+  )
+  expect_equal(
+    recipes_eval_select(quos = quos(date:age), data = okc, info = info1),
+    setNames(nm = c("date", "location", "height", "diet", "age"))
+  )
+  expect_equal(
+    recipes_eval_select(quos = quos(matches("blahblahblah")), data = okc, info = info1),
+    setNames(nm = character())
+  )
+
+  expect_error(
+    recipes_eval_select(quos = quos(log(date)), data = okc, info = info1),
+    "non-numeric argument to mathematical function"
+  )
+  expect_error(
+    recipes_eval_select(quos = quos(I(date:age)), data = okc, info = info1),
+    "object 'age' not found"
+  )
+  expect_error(
+    recipes_eval_select(data = okc, info = info1),
+    'argument "quos" is missing, with no default'
+  )
```

Comment:
You should try using `expect_snapshot(error = TRUE, {expression})` here. Expecting an exact match against an error that you don't own is fragile, and `expect_snapshot()` won't fail on CRAN if this error ever changes (doubtful that it would, but still)